<launch>
    <node pkg="comp_strategy" type="strategy" name="comp2018_core" output="screen">

        <param name="REFRESH_RATE"          value="10"/>

        <param name="HEIGHT_MAX"            value="3.0"/>
        <param name="SPD_Z_MAX"             value="0.4"/>
        <param name="TOLLERANCE_Z"          value="0.1"/>
        <param name="SPD_Z_SMALL"           value="0.15"/>
        <param name="SAFE_RANGE_Z"          value="0.4"/>

        <param name="SPD_XY_MAX"				value="0.4"/>
        <param name="TOLLERANCE_XY"             value="0.133"/>
        <param name="SPD_XY_SMALL"              value="0.2"/>
        <param name="SAFE_RANGE_XY"             value="1.35"/>

        <param name="CRUISE_HEIGHT_CIRCLE"                 value="0.75"/>

        <param name="QR_OFFSET23"                   value="0.0"/>
        <param name="QR_GLOBOL_OFFSET_X"            value="0"/>
        <param name="QR_GLOBOL_OFFSET_Y"            value="0"/>
        <param name="X_RESTRICT_LEFT"               value="-7"/>
        <param name="X_RESTRICT_RIGHT"              value="7"/>
        <param name="SIMPLE_DETECT_OFFSET"                 value="1.0"/>

        <param name="CIRCLE_SEARCH_OFFSET_Y"           value="1"/>
        <param name="CIRCLE_SEARCH_OFFSET_Z"           value="0.2"/>
        <param name="CIRCLE_PASS_DIST"                 value="1.0"/>
        <param name="LOW_CIRCLE_BOARD_HEIGHT"          value="0.45"/>
        <param name="HIGH_CIRCLE_BOARD_HEIGHT"         value="0.75"/>
        <param name="FIRST_RANGE_BOARD_AIM"            value="0.4"/>
        <param name="CIRCLE_BOARD_AIM_TOLLERANCE"      value="0.1"/>
        <param name="AIM_BOARD_P"                      value="0.75"/>
        <param name="AIM_BOARD_SATISFIED_COUNT"        value="2"/>
        <param name="CIRCLE_CLIMB_SLEEP_TIME"          value="1"/>

        <param name="CRUISE_HEIGHT_BOARD"         value="1.3"/>
        <param name="BOARD_AIM_TOLLERANCE"     value="0.3"/>
        <param name="BOARD_AIM_HEIGHT"     value="1.3"/>

        <param name="QR_SEARCH_OFFSET"                  value="1.5"/>
        <param name="QR_TOLLANCE"                       value="0.3"/>
        <param name="QR_CRUISE_HEIGHT"                  value="1.15"/>

        <param name="DYNAMIC_DETECT_ENABLE"                 value="true"/>
        <param name="DYNAMIC_DETECT_TOLLERANCE"             value="0.5"/>
        <param name="MATCH_DETECT_TOLLERANCE"               value="0.6"/>

        <param name="THRESHOLD_CAM_BOARD_FIRST"               value="0.5"/>
        <param name="THRESHOLD_CAM_BOARD_SECOND"               value="2.0"/>
        <param name="THRESHOLD_CAM_CIRCLE_FIRST"               value="0.75"/>
        <param name="THRESHOLD_CAM_CIRCLE_SECOND"               value="2.0"/>
        <param name="THRESHOLD_CAM_TREE_FIRST"               value="0.75"/>
        <param name="THRESHOLD_CAM_TREE_SECOND"               value="2.0"/>
        <param name="THRESHOLD_CAM_QR_SECOND"               value="2.25"/>

        <param name="BOTH_SAFETY_CHECK_TIME"               value="0.45"/>

        <param name="FINAL_X"               value="1.5"/>
        <param name="FINAL_Y"               value="1.5"/>

        <param name="DETECT_RANGE_X_LEFT"               value="-6"/>
        <param name="DETECT_RANGE_X_RIGHT"               value="6"/>

        <param name="BEGIN_NUM"               value="1"/>
    </node>
</launch>